### ISSUE TITLE
Add a visually hidden label to the search block

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#34989]
## What problem does this address?

The search block has an option to toggle the form label on and off.
Turning it off removes the label completely. This results in a lack of context for non sighted users.

Read more:
https://github.com/WordPress/gutenberg/pull/24666#issuecomment-687835520
https://dequeuniversity.com/rules/axe/3.3/label

## What is your proposed solution?

Instead of removing it completely, a visually hidden label should be present.
The visually hidden form label should work both with a custom text and the default label.
